great_expectations/metrics/domain.py

```diff
@@ -59,6 +59,21 @@ class ColumnValues(Values):
     column: NonEmptyString
 
 
+class ColumnPairValues(Values):
+    """A domain type for metrics that compute row-level calculations on multiple columns.
+
+    The ColumnPairValues domain type is used to define metrics that evaluate conditions or compute
+    values for each row in multiple columns. This class is intended to be used as a mixin
+    with the Metric class when defining a new Metric.
+    """
+
+    column_A: NonEmptyString
+    column_B: NonEmptyString
+    ignore_row_if: Literal["both_values_are_missing", "either_value_is_missing", "neither"] = (
+        "both_values_are_missing"
+    )
+
+
```

Comment:
You will need to add `column_A`, `column_B`, and `ignore_row_if` to
```
class ColumnPairMetric(BatchMetric[_MetricResult]): ...
```
in `great_expectations/metrics/column_pair/column_pair.py` and add the class to the `great_expectations/metrics/column_pair` package to shorten the import.